tests.ts

```diff
@@ -0,0 +1,67 @@
+const functsFromMain = require('./main');
+const assert = require('assert');
+
+// test for formReadRes function
+describe("formReadRes", () => {
+    // execute function
+    functsFromMain.formReadRes('data.txt')('result.txt')("l")
```

Comment:
This is a function call which executes asynchronous function, i.e. function, which takes time to execute. Ignoring that, you  execute it outside the test, but in the code of the test do not attempt to wait for it to complete. Instead, you start to readFile immediately. Meaning, `formReadRes` is probably not ready by the time you start running the test code.